Rocky.agent.lib.nut

```diff
@@ -0,0 +1,1084 @@
+        routeHandler.setTimeout(timeout);
+        _handlers[signature][verb] <- routeHandler;
+        return routeHandler;
+    },
+
+    /**
+     * Register a handler for an HTTP POST request.
+     *
+     * @param {string}   signature - An endpoint path signature.
+     * @param {function} callback  - The handler that will process the POST request.
+     * @param {integer}  timeout   - Optional timeout in seconds. Default: the class-level value.
+     *
+     * @returns {object} A Rocky.Route instance for the handler.
+    */
+    "post": function(signature, callback, timeout=null) {
+        return on("POST", signature, callback, timeout);
+    },
+
+    /**
+     * Register a handler for an HTTP GET request.
+     *
+     * @param {string}   signature - An endpoint path signature.
+     * @param {function} callback  - The handler that will process the GET request.
+     * @param {integer}  timeout   - Optional timeout in seconds. Default: the class-level value.
+     *
+     * @returns {object} A Rocky.Route instance for the handler.
+    */
+    "get": function(signature, callback, timeout=null) {
+        return on("GET", signature, callback, timeout);
+    },
+
+    /**
+     * Register a handler for an HTTP PUT request.
+     *
+     * @param {string}   signature - An endpoint path signature.
+     * @param {function} callback  - The handler that will process the PUT request.
+     * @param {integer}  timeout   - Optional timeout in seconds. Default: the class-level value.
+     *
+     * @returns {object} A Rocky.Route instance for the handler.
+    */
+    "put": function(signature, callback, timeout=null) {
+        return on("PUT", signature, callback, timeout);
+    },
+
+    // ------------------- AUTHORIZATION -------------------//
+
+    /**
+     * Register a handler for request authorization.
+     *
+     * @param {Function} callback - The handler that will process authorization requests.
+     *
+     * @returns {object} The Rocky instance (this).
+    */
+    "authorize": function(callback) {
+        _handlers.authorize <- callback;
+        return this;
+    },
+
+    /**
+     * Register a handler for processing rejected requests.
+     *
+     * @param {function} callback - The handler that will process rejected requests.
+     *
+     * @returns {object} The Rocky instance (this).
+    */
+    "onUnauthorized": function(callback) {
+        _handlers.onUnauthorized <- callback;
+        return this;
+    },
+
+    // -------------------      EVENTS    -------------------//
+
+    /**
+     * Register a handler for timed out requests.
+     *
+     * @param {function} callback - The handler that will process request time-outs.
+     * @param {integer/float}  timeout  - Optional timeout in seconds. Default: the class-level value.
+     *
+     * @returns {object} The Rocky instance (this).
+    */
+    "onTimeout": function(callback, timeout = null) {
+        if (timeout == null) timeout = _timeout;
```

Comment:
It appears the timeout parameter is just dropped here? How is this parameter supposed to be used? The previous version updated the global timeout setting with the one passed in.

Reply:
Good spot. I may have deleted that line while un-refactoring.